src/lib/data/analytics.js

```diff
@@ -0,0 +1,116 @@
+import Analytics from 'analytics-node';
+import { isStyledComponent } from "styled-components";
+import { nanoid } from 'nanoid';
+import { debounce, get, split, truncate } from 'lodash'
+
+let segment
+let anonymousUserId
+let userIdCreatedAt
+
+function getAnonymousId() {
+  if (anonymousUserId) {
+    return anonymousUserId;
+  }
+
+  const storageId = localStorage.getItem('anonymousUserId');
+  userIdCreatedAt = localStorage.getItem('anonymousUserIdCreatedAt');
+
+  if (storageId) {
+    anonymousUserId = storageId;
+  } else {
+    anonymousUserId = nanoid();
+    userIdCreatedAt = new Date().toUTCString()
+    localStorage.setItem('anonymousUserId', anonymousUserId);
+    localStorage.setItem('anonymousUserIdCreatedAt', userIdCreatedAt);
+  }
+
+  return anonymousUserId;
+}
+
+export function init() {
+  if (segment) return; // already initialized
+
+  getAnonymousId();
+  try {
+    segment = new Analytics("diA7hiO28gGeb9fxn615Xs91uX3GyYhL", {});
```

Comment:
is this hardcoding such that everyone who uses the open source VM package is automatically sending usage analytics into our Segment account?

Reply:
That's correct. I'm open to setting this as a github variable, to support keeping develop and production events separate, but that would mean individuals who fork this repo would have to update this code manually.

Reply:
We may also want to make improvements over the long-term such as using a proxy to avoid ad blockers.

Reply:
I am more concerned about the optics of centralized usage tracking in a decentralized project. I would expect each gateway to have the power to set where/if analytics are sent

Reply:
I don't feel comfortable making the final call on this, at least not in this repo (maybe if we fork). I will defer to @evgenykuzyakov 

Ideally I would want the Segment ID to be passed into the VM— making it opt-in on a per-gateway basis

Reply:
Here would be an example of we can pass in the config to the VM's existing  `useInitNear()` hook (`segmentId` would be optional - if a value isn't passed, segment would not be initialized):

```jsx
import { useInitNear } from "near-social-vm";

const { initNear } = useInitNear();

useEffect(() => {
    initNear &&
      initNear({
        segmentId: '...'
        networkId: '...',
        ...
      });
  }, [initNear]);
```